Update makefile: for easy to debug

Sn0rt opened the pull request:
Signed-off-by: Sn0rt <Sn0rt@abc.shop.edu.cn>

It can be easy to debug if Makefile can import GOFLAG environment variable.

**go build flag**
```shell
$ make GOFLAG="-v" tools
go build -v -ldflags "-X main.gitCommit=0a1e5dc0c9b67619d46bd8cbcc948bda1b143b27" ./cmd/oci-create-runtime-bundle
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/specs-go
github.com/opencontainers/image-tools/vendor/github.com/pkg/errors
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonpointer
github.com/opencontainers/image-tools/vendor/go4.org/errorutil
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/specs-go/v1
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/runtime-spec/specs-go
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonreference
github.com/opencontainers/image-tools/vendor/github.com/spf13/pflag
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonschema
github.com/opencontainers/image-tools/vendor/github.com/spf13/cobra
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/schema
github.com/opencontainers/image-tools/image
github.com/opencontainers/image-tools/cmd/oci-create-runtime-bundle
go build -v -ldflags "-X main.gitCommit=0a1e5dc0c9b67619d46bd8cbcc948bda1b143b27" ./cmd/oci-image-validate
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/specs-go
github.com/opencontainers/image-tools/vendor/github.com/pkg/errors
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonpointer
github.com/opencontainers/image-tools/vendor/go4.org/errorutil
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/specs-go/v1
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/runtime-spec/specs-go
github.com/opencontainers/image-tools/vendor/github.com/spf13/pflag
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonreference
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonschema
github.com/opencontainers/image-tools/vendor/github.com/spf13/cobra
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/schema
github.com/opencontainers/image-tools/image
github.com/opencontainers/image-tools/cmd/oci-image-validate
go build -v -ldflags "-X main.gitCommit=0a1e5dc0c9b67619d46bd8cbcc948bda1b143b27" ./cmd/oci-unpack
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/specs-go
github.com/opencontainers/image-tools/vendor/github.com/pkg/errors
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonpointer
github.com/opencontainers/image-tools/vendor/go4.org/errorutil
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/specs-go/v1
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/runtime-spec/specs-go
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonreference
github.com/opencontainers/image-tools/vendor/github.com/spf13/pflag
github.com/opencontainers/image-tools/vendor/github.com/xeipuuv/gojsonschema
github.com/opencontainers/image-tools/vendor/github.com/spf13/cobra
github.com/opencontainers/image-tools/vendor/github.com/opencontainers/image-spec/schema
github.com/opencontainers/image-tools/image
github.com/opencontainers/image-tools/cmd/oci-unpack
```

**go test flag**
```shell
$ make GOFLAG="-v" test
go test -race -cover -v github.com/opencontainers/image-tools/cmd/oci-create-runtime-bundle github.com/opencontainers/image-tools/cmd/oci-image-validate github.com/opencontainers/image-tools/cmd/oci-unpack github.com/opencontainers/image-tools/image
?   	github.com/opencontainers/image-tools/cmd/oci-create-runtime-bundle	[no test files]
?   	github.com/opencontainers/image-tools/cmd/oci-image-validate	[no test files]
?   	github.com/opencontainers/image-tools/cmd/oci-unpack	[no test files]
=== RUN   TestValidateLayout
--- PASS: TestValidateLayout (0.02s)
=== RUN   TestUnpackLayerDuplicateEntries
--- PASS: TestUnpackLayerDuplicateEntries (0.01s)
=== RUN   TestUnpackLayer
--- PASS: TestUnpackLayer (0.01s)
=== RUN   TestUnpackLayerRemovePartialyUnpackedFile
--- PASS: TestUnpackLayerRemovePartialyUnpackedFile (0.01s)
=== RUN   TestNewTarWalker
--- PASS: TestNewTarWalker (0.00s)
PASS
coverage: 34.5% of statements
ok  	github.com/opencontainers/image-tools/image	1.054s	coverage: 34.5% of statements
```

Sn0rt commented:
@xiekeyang Can you help me that review this PR?

xiekeyang commented:
I've read this patch, but I doubt if it is useful. 
Current output seems enough to show result and root cause(if failure).
`v` (print the names of packages as they are compiled (or test)) seems not what we want to debug.
I feel This modification doesn't bring obvious benefit for debug. 
ping @wking

wking commented:
On Thu, Nov 10, 2016 at 12:39:35AM -0800, xiekeyang wrote:
> I feel This modification doesn't bring obvious benefit for debug. 

This is where I'm at too.  I'm more interested in the failure messages
(which we get with or without -v) than about the success messages.
But I don't feel strongly about it.  If folks want to land this PR,
that's fine with me.

Sn0rt commented:
The "-v" argument is weak but which is useful at least, maybe  need't to modify .travis.yml.

**EXAMPLE IN REAL WORLD:**
```shell
$ govendor update +v github.com/opencontainers/ 
$ git st
On branch update_makefile
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   vendor/github.com/opencontainers/image-spec/schema/fs.go
	modified:   vendor/github.com/opencontainers/image-spec/schema/validator.go
	deleted:    vendor/github.com/opencontainers/image-spec/specs-go/descriptor.go
	modified:   vendor/github.com/opencontainers/image-spec/specs-go/v1/config.go
	modified:   vendor/github.com/opencontainers/image-spec/specs-go/v1/manifest.go
	modified:   vendor/github.com/opencontainers/image-spec/specs-go/v1/manifest_list.go
	modified:   vendor/github.com/opencontainers/image-spec/specs-go/v1/mediatype.go
	modified:   vendor/github.com/opencontainers/image-spec/specs-go/version.go
	modified:   vendor/github.com/opencontainers/image-spec/specs-go/versioned.go

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	vendor/github.com/opencontainers/image-spec/schema/config-schema.json
	vendor/github.com/opencontainers/image-spec/schema/config_test.go
	vendor/github.com/opencontainers/image-spec/schema/content-descriptor.json
	vendor/github.com/opencontainers/image-spec/schema/defs-config.json
	vendor/github.com/opencontainers/image-spec/schema/defs-image.json
	vendor/github.com/opencontainers/image-spec/schema/defs.json
	vendor/github.com/opencontainers/image-spec/schema/image-manifest-schema.json
	vendor/github.com/opencontainers/image-spec/schema/manifest-list-schema.json
	vendor/github.com/opencontainers/image-spec/schema/manifest_backwards_compatibility_test.go
	vendor/github.com/opencontainers/image-spec/schema/manifest_test.go
	vendor/github.com/opencontainers/image-spec/schema/spec_test.go
	vendor/github.com/opencontainers/image-spec/specs-go/v1/descriptor.go
	vendor/github.com/opencontainers/image-spec/specs-go/v1/layout.go
	vendor/vendor.json

no changes added to commit (use "git add" and/or "git commit -a")
$ make test
go test -race -cover  github.com/opencontainers/image-tools/cmd/oci-create-runtime-bundle github.com/opencontainers/image-tools/cmd/oci-image-validate github.com/opencontainers/image-tools/cmd/oci-unpack github.com/opencontainers/image-tools/image
?   	github.com/opencontainers/image-tools/cmd/oci-create-runtime-bundle	[no test files]
?   	github.com/opencontainers/image-tools/cmd/oci-image-validate	[no test files]
?   	github.com/opencontainers/image-tools/cmd/oci-unpack	[no test files]
warning: layer sha256:f0627cc952bec47168ba4397d73ad2cf4022d439c2e2cc96d8b0d632506c4055 has an unknown media type: application/vnd.oci.image.layer.tar+gzip
--- FAIL: TestValidateLayout (0.02s)
	image_test.go:156: layer validation failed: invalid descriptor MediaType "application/vnd.oci.image.layer.tar+gzip"
--- FAIL: TestUnpackLayer (0.01s)
	manifest_test.go:122: stat /tmp/test-layer624244301/rootfs/test: no such file or directory
FAIL
coverage: 31.0% of statements
FAIL	github.com/opencontainers/image-tools/image	0.055s
Makefile:38: recipe for target 'test' failed
make: *** [test] Error 1

$ make GOFLAG="-v" test
go test -race -cover -v github.com/opencontainers/image-tools/cmd/oci-create-runtime-bundle github.com/opencontainers/image-tools/cmd/oci-image-validate github.com/opencontainers/image-tools/cmd/oci-unpack github.com/opencontainers/image-tools/image
?   	github.com/opencontainers/image-tools/cmd/oci-create-runtime-bundle	[no test files]
?   	github.com/opencontainers/image-tools/cmd/oci-image-validate	[no test files]
?   	github.com/opencontainers/image-tools/cmd/oci-unpack	[no test files]
=== RUN   TestValidateLayout
warning: layer sha256:f0627cc952bec47168ba4397d73ad2cf4022d439c2e2cc96d8b0d632506c4055 has an unknown media type: application/vnd.oci.image.layer.tar+gzip
--- FAIL: TestValidateLayout (0.02s)
	image_test.go:156: layer validation failed: invalid descriptor MediaType "application/vnd.oci.image.layer.tar+gzip"
=== RUN   TestUnpackLayerDuplicateEntries
--- PASS: TestUnpackLayerDuplicateEntries (0.01s)
=== RUN   TestUnpackLayer
--- FAIL: TestUnpackLayer (0.01s)
	manifest_test.go:122: stat /tmp/test-layer858078967/rootfs/test: no such file or directory
=== RUN   TestUnpackLayerRemovePartialyUnpackedFile
--- PASS: TestUnpackLayerRemovePartialyUnpackedFile (0.01s)
FAIL
coverage: 31.0% of statements
FAIL	github.com/opencontainers/image-tools/image	0.048s
Makefile:38: recipe for target 'test' failed
make: *** [test] Error 1
```

xiekeyang commented:
Just my personal opinion: `Makefile`, being way to organize code compilation, should bring mainly the help info, result of compilation, test, lint..., and the message where the error happen. `-v` provides the package name(compiling) and function name(test). To get that(seldom), I rather run `go test -v` directly, than add all useful but not necessary stuff to `Makefile`.
Above is my feeling but I'm not opposed other folks to land it.

Sn0rt commented:
@xiekeyang Maybe you are right, the makefile should not include this code.
